File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        String str=String.valueOf(n);
        str=str.replaceAll("0","");
        if (str.isEmpty()) return 0;
        long num=Long.parseLong(str);
        long s=0;
        for(int i=0;i<str.length();i++){
            s+=str.charAt(i)-'0';
        }
        return num*s;
    }
}

File: 3754-concatenate-non-zero-digits-and-multiply-by-sum-i/3754-concatenate-non-zero-digits-and-multiply-by-sum-i.cpp
class Solution {
public:
    long long sumAndMultiply(int n) {
        long long sum=0;
        long long ans=0;
        int p=1;
        while(n!=0){
            int r=n%10;
            n/=10;
            if(r==0) continue;
            sum+=r;
            ans+=(r*p);
            p*=10;
        }
        return ans*sum;
    }
};